model/CMakeLists.txt

```diff
@@ -478,28 +501,25 @@ foreach (xlen IN ITEMS 32 64)
                     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
                     COMMAND
                         ${SAIL_BIN}
-                        --lean
-                        --memo-z3
-                        --require-version ${SAIL_REQUIRED_VER}
-                        --lean-output-dir ${CMAKE_CURRENT_BINARY_DIR}
-                        --lean-force-output
-                        --lean-noncomputable
-                        --lean-noncomputable-function encdec_forwards
-                        --lean-noncomputable-function encdec_backwards
-                        --lean-noncomputable-function encdec_forwards_matches
-                        --lean-noncomputable-function encdec_backwards_matches
-                        --lean-noncomputable-function encdec_compressed_forwards
-                        --lean-noncomputable-function encdec_compressed_backwards
-                        --lean-noncomputable-function encdec_compressed_forwards_matches
-                        --lean-noncomputable-function encdec_compressed_backwards_matches
-                        --lean-non-beq-type ast
-                        --lean-import-file ../handwritten_support/RiscvExtras.lean
-                        --config ${CMAKE_SOURCE_DIR}/config/default.json
-                        -o "Lean_${arch_uppercase}"
+                        ${lean_sail_common}
+                        ${lean_sail_default}
+                        ${sail_srcs}
+                )
+                add_custom_command(
+                    DEPENDS ${sail_srcs}
```

Comment:
This is missing some dependencies - at least `config/default.json` and `handwritten_support/RiscvExtrasExecutable.lean` and `handwritte_supports/RiscvExtras.lean`. It means incremental builds won't work properly if you edit those files. Same for line 497.

Existing problem though so don't feel like you need to fix it in this PR - just pointing it out.

Reply:
I fixed it in this PR since it's a quick change. Thank you for the remark!